File: backend/internal/model/team.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type TeamMemberRole string

const (
	TeamMemberRoleAdmin      TeamMemberRole = "admin"
	TeamMemberRoleMaintainer TeamMemberRole = "maintainer"
	TeamMemberRoleViewer     TeamMemberRole = "viewer"
)

type TeamMemberStatus string

const (
	TeamMemberStatusPending  TeamMemberStatus = "pending"
	TeamMemberStatusApproved TeamMemberStatus = "approved"
	TeamMemberStatusRejected TeamMemberStatus = "rejected"
)

type Team struct {
	ID                      uuid.UUID  `db:"id" json:"id"`
	Name                    string     `db:"name" json:"name"`
	Slug                    string     `db:"slug" json:"slug"`
	IsDiscoverable          bool       `db:"is_discoverable" json:"is_discoverable"`
	CreatedBy               *uuid.UUID `db:"created_by" json:"created_by"`
	GithubPATEncrypted      *string    `db:"github_pat_encrypted" json:"-"`
	GithubPATNonce          []byte     `db:"github_pat_nonce" json:"-"`
	GithubPATSetAt          *time.Time `db:"github_pat_set_at" json:"-"`
	GithubPATSetBy          *uuid.UUID `db:"github_pat_set_by" json:"-"`
	GithubPATGithubUsername *string    `db:"github_pat_github_username" json:"-"`
	CreatedAt               time.Time  `db:"created_at" json:"created_at"`
	UpdatedAt               time.Time  `db:"updated_at" json:"updated_at"`
	DeletedAt               *time.Time `db:"deleted_at" json:"-"`
}

func (t *Team) HasPAT() bool {
	return t.GithubPATEncrypted != nil && *t.GithubPATEncrypted != ""
}

type TeamMember struct {
	ID        uuid.UUID        `db:"id" json:"id"`
	TeamID    uuid.UUID        `db:"team_id" json:"team_id"`
	UserID    uuid.UUID        `db:"user_id" json:"user_id"`
	Role      TeamMemberRole   `db:"role" json:"role"`
	Status    TeamMemberStatus `db:"status" json:"status"`
	InvitedBy *uuid.UUID       `db:"invited_by" json:"invited_by"`
	CreatedAt time.Time        `db:"created_at" json:"created_at"`
}

type TeamJoinRequest struct {
	ID           uuid.UUID        `db:"id" json:"id"`
	TeamID       uuid.UUID        `db:"team_id" json:"team_id"`
	UserID       uuid.UUID        `db:"user_id" json:"user_id"`
	Message      *string          `db:"message" json:"message"`
	Status       TeamMemberStatus `db:"status" json:"status"`
	ReviewedBy   *uuid.UUID       `db:"reviewed_by" json:"reviewed_by"`
	RoleAssigned *TeamMemberRole  `db:"role_assigned" json:"role_assigned"`
	CreatedAt    time.Time        `db:"created_at" json:"created_at"`
	UpdatedAt    time.Time        `db:"updated_at" json:"updated_at"`
}

File: frontend/src/types/team.ts
export type TeamRole = "admin" | "maintainer" | "viewer";
export type MemberStatus = "pending" | "approved" | "rejected";

export interface Team {
  id: string;
  name: string;
  slug: string;
  my_role?: TeamRole;
  member_count: number;
  repo_count: number;
  created_at: string;
  updated_at: string;
}

export interface TeamDetail extends Team {
  members: TeamMember[];
  repositories: TeamRepo[];
}

export interface TeamMember {
  id: string;
  user_id: string;
  username: string;
  email: string;
  avatar_url: string | null;
  role: TeamRole;
  status: MemberStatus;
  invited_by: string | null;
  created_at: string;
}

export interface TeamJoinRequest {
  id: string;
  user: {
    id: string;
    username: string;
    email: string;
    avatar_url: string | null;
  };
  message: string | null;
  status: MemberStatus;
  role_assigned: TeamRole | null;
  reviewed_by: string | null;
  created_at: string;
  updated_at: string;
}

export interface TeamRepo {
  id: string;
  team_id: string;
  repository_id: string;
  repository_name: string;
  github_full_name: string;
  is_active: boolean;
  added_by: string | null;
  created_at: string;
}

export interface DiscoverableTeam {
  id: string;
  name: string;
  slug: string;
  member_count: number;
  repo_count: number;
  created_at: string;
  user_status: "pending" | "approved" | null;
}
